doc/ch-grow_desc.rst

```diff
@@ -3,89 +3,146 @@ Synopsis
 
 ::
 
-   $ ch-grow [OPTIONS] [-t TAG] [-f DOCKERFILE] CONTEXT
+   $ ch-grow [...] build [-t TAG] [-f DOCKERFILE] [...] CONTEXT
+   $ ch-grow [...] print-storage
+   $ ch-grow [...] pull [...] IMAGE_REF [IMAGE_DIR]
+   $ ch-grow { --help | --version | --dependencies }
```

Comment:
I think this is a step in the right direction. However, I think we would eventually want to abstract yet another layer. See below.

```
   $ ch [...] build [-b|--builder] [-t TAG] [-f DOCKERFILE] [...] CONTEXT
   $ ch [...] list [...] # e.g., print-storage, images, etc.
   $ ch [...] pull [...] IMAGE_REF [IMAGE_DIR]
   $ ch [...] push [...] IMAGE_REF DEST_IMG_REF
   $ ch [...] run [OPTIONS] IMAGE -- CMDS
   $ ch { --help | --version | --dependencies }
```

We have a lot of manual tools right now. Condensing some of them will dramatically change the workflow. How annoying would it be to have the interface changed (admittedly for the better) just to be changed again down the road?

Reply:
This is a good point, though I'm still not convinced we want to do subcommands across all of Charliecloud. In particular I see running vs. building/pushing/pulling/etc. as separate operations.

As for doing it right now, I'm seeing maintainability problems already. E.g. `ch-tug` has a bunch of duplicate code and the command line options have diverged from `ch-grow`, and we have a bunch of bug reports titled "ch-grow/ch-tug: blah blah blah"; the latter doesn't scale to `ch-nudge` at all. Why does `ch-grow --list` exist but `ch-nudge` is a separate script? Etc.